src/sentry/api/endpoints/organization_group_index.py

```diff
@@ -118,16 +118,18 @@ def get(self, request, organization):
             # check to see if we've got an event ID
             event_id = normalize_event_id(query)
             if event_id:
-                groups = list(
-                    Group.objects.filter_by_event_id(project_ids, event_id)
+                # For a direct hit lookup we want to use any passed project ids
+                # (we've already checked permissions on these) plus any other
+                # projects that the user is a member of. This gives us a better
+                # chance of returning the correct result, even if the wrong
+                # project is selected.
+                direct_hit_projects = set(project_ids) | set(
+                    [project.id for project in request.access.projects]
```

Comment:
Interesting - I haven't seen this before.. normally we do `self.get_projects()`. Is there a reason we're accessing the projects in a different way here

Reply:
This uses `self.get_projects` (take a look where `project_ids` comes from), it just supplements them with all the projects you have access to for this lookup.